api/v1beta1/cluster_types.go

```diff
@@ -772,18 +778,21 @@ type MachinePoolTopology struct {
 	// class is the name of the MachinePoolClass used to create the pool of worker nodes.
 	// This should match one of the deployment classes defined in the ClusterClass object
 	// mentioned in the `Cluster.Spec.Class` field.
+	// +kubebuilder:validation:MaxLength=256
```

Comment:
253 max, min 1, required name of a custom resource

Reply:
I'll add min length, otherwise same answer as: https://github.com/kubernetes-sigs/cluster-api/pull/11949#discussion_r1987800527

Reply:
(I'l resolve to deduplicate)